apps/gatsby/src/AppConfig/AppConfig.js

```diff
@@ -150,26 +157,80 @@ export class AppConfig extends React.Component {
     }
   };
 
-  toggleContentType = (enabledContentTypes, ctId) => {
-    if (enabledContentTypes.includes(ctId)) {
-      return enabledContentTypes.filter((cur) => cur !== ctId);
+  selectorTypeToggle = () => {
+    this.setState((prevState) => ({
+      selectorType: !prevState.selectorType
+    }))
+  }
+
+  disableContentType = (id) => {
+    const newEnabledTypes = this.state.enabledContentTypes.filter(type => type !== id);
+    const shouldUpdate = this.state.enabledContentTypes.length > newEnabledTypes.length
+    if (shouldUpdate) {
+      this.setState(() => (
+        {
+          enabledContentTypes: newEnabledTypes
+        }
+      ))
+    }
+  }
+
+  toggleContentType = (enabledContentTypes, newId, prevId) => {
+    if (enabledContentTypes.includes(prevId) && prevId !== newId) {
+      //Swap in the new id at the correct index in state to avoid the movement in the UI
+      const index = enabledContentTypes.findIndex(id => id === prevId);
+      enabledContentTypes[index] = newId;
+      return enabledContentTypes;
     } else {
-      return enabledContentTypes.concat([ctId]);
+      return enabledContentTypes.concat([newId]);
     }
   };
 
-  onContentTypeToggle = (ctId) => {
+  onContentTypeToggle = (newId, prevId) => {
     this.setState((prevState) => ({
       ...prevState,
       enabledContentTypes: this.toggleContentType(
         prevState.enabledContentTypes,
-        ctId
+        newId,
+        prevId
       ),
     }));
   };
 
+  updateUrlConstructors = (currentUrlConstructors, id, newInput) => {
+    let constructors;
+    // Check if the constructor needs to be added, or if an id that already exists needs a new slug
+    const index = currentUrlConstructors.findIndex(cur => cur.id === id)
+    if (index !== -1) {
+      currentUrlConstructors[index].slug = newInput
+       constructors = currentUrlConstructors
+    } else {
+      const newConstructor = {
+        id,
+        slug: newInput
+      }
+      constructors = [...currentUrlConstructors, ...[newConstructor]]
```

Comment:
Maybe we should add some validation logic here. Tried to use the feature, but it wasn't super clear to me why things weren't working.

As a matter of fact, I have only managed to get the "string" case to work before reading the code. The rest did not and without notifying me of why (e.g. was the input in incorrect format?)

Reply:
I don't know how useful it would be. As far as I know, the only thing that I could validate is that a string is being passed. Whether that string results in the desired value is decided in the sidebar.js file. I recognize that this part of the setup is a bit confusing though. I think a good solution could be creating a minute long explainer video, that I'll link in the markup, so people have a better idea of what is going on.

Reply:
Added an explainer video